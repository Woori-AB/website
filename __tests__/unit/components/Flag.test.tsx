import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { describe, it, vi, expect, afterEach } from "vitest";

describe("Flag", () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  it("matches snapshot(s)", async () => {
    // Arrange
    const { default: Flag } = await vi.importActual<
      // eslint-disable-next-line @typescript-eslint/consistent-type-imports
      typeof import("@/components/Flag")
    >("@/components/Flag");

    // Act
    const { container } = render(
      <Flag src="/flags/sv.svg" alt="Svenska" onClick={() => null} />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });

  it("invokes the onClick callback when clicked", async () => {
    // Arrange
    const { default: Flag } = await vi.importActual<
      // eslint-disable-next-line @typescript-eslint/consistent-type-imports
      typeof import("@/components/Flag")
    >("@/components/Flag");

    const onClick = vi.fn();

    // Act
    render(<Flag src="/flags/sv.svg" alt="Svenska" onClick={onClick} />);
    const flag = screen.getByAltText("Svenska");
    fireEvent.click(flag);

    // Assert
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
