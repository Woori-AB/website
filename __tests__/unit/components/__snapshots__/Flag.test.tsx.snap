// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flag > matches snapshot(s) 1`] = `
<div>
  <img
    alt="Svenska"
    class="h-5 w-auto block"
    src="/flags/sv.svg"
    title="Svenska"
  />
</div>
`;
